evennia/utils/evmenu.py

```diff
@@ -64,10 +64,63 @@ def another_node(caller, input_string, **kwargs):
 The return values must be given in the above order, but each can be
 returned as None as well. If the options are returned as None, the
 menu is immediately exited and the default "look" command is called.
+    text (str, tuple, dict or None): Text shown at this node. If a dict
+        it can also contain various formatting instructions. In such a
+        case the following keys are already supported:
+
+            - 'header' (dict): Dictionary of header data for the node
+            - 'text' (dict): Dictionary of text data for the node
+            - 'options' (dict): Dictionary of options data for the node
+            - 'footer' (dict): Dictionary of footer data for the node
+            - 'help' (dict): Dictionary of help data for the node
+
+            Each of these keys support their own dictionary with the
+            following keys:
+
+                - 'contents' (string or list): Required. String data for
+                    all keys other than 'options'. List in the case of
+                    options.
+                - 'formatter' (string): Optional. Designator of the
+                    function used to format the contents. Unless this
+                    code has been subclassed the only options are 'bars'
+                    or 'no_bars'.
+
+            In the case of 'options' the following additional keys are
+            supported by its dictionary:
+
+                - 'hidekeys' (list): A list of keys. The options for
+                    those keys will not be listed in the options table
+                    though the node will accept them as input.
+                - 'hidedescs' (list): As hidekey but using the desc of
+                    the options instead.
+                - 'movekeys' (list): A list of keys. The options for
+                    those keys will be put into a separate list after
+                    the options table.
+                - 'movedescs' (list): As movekeys but using the desc of
+                    the options instead.
+                - 'rows' (int): Will attempt to make the options table
+                    the specified number of rows.
+
+            - 'node' (dict): A dict for consistency of use. At present
+                it only handles a single key:
+
+                - 'formatter' (str): String designating the function to
+                    be used for final assembly of the node. Unless this
+                    code has been subclasses it currently supports three
+                    options:
+                    - 'suppress': The node is not drawn at all
+                    - 'default': The node is assembled similarly to the
+                        legacy method.
+                    - 'invert': The node is assembled with the header
+                        followed by the options table, the general text,
+                        and then the footer, more or less inverting the
+                        traditional layout.
```

Comment:
It's unclear where this block belongs ... I presume it belongs in the outermost level (next to `options`, `text` etc), but not sure a separate `node` level is needed. Just have `formatter` at the top level next to `options`, `text` ...) - by its placement I think this would be quite clearly understood as a top-level formatter (and its arguments) that will affect every node.

Reply:
I didn't want to refer to it purely as ``formatter`` because I'm concerned that it will become confused with the ``formatters`` for the individual entries, so ``node`` seemed a better option. I also realized that leaving it as a ``dict`` seems like a better choice because it makes it possible for someone to pass variables to a custom node formatter if they want to.

Reply:
"node" is yet another custom key to remember. Better to use the same form as elsewhere: 
- `formatter = callable_or_str`
- `formatter = (callable_or_str, {kwargs})`

Reply:
I see what you're saying there now. Essentially the entire dict is the 'node' dict so the 'formatter' option would be for the entire body and everything else are components used by that formatter. The each component has its own formatter option along with their specific components.